lib/src/r_tree/non_leaf_node.dart

```diff
@@ -119,31 +115,9 @@ class NonLeafNode<E> extends Node<E> {
     return bestNode;
   }
 
-  _convertToLeafNode() {
-    var nonLeafParent = parent as NonLeafNode<E>?;
-    if (nonLeafParent == null) return;
-
-    var newLeafNode = LeafNode<E>(this.branchFactor);
-    nonLeafParent.removeChild(this);
-    nonLeafParent.addChild(newLeafNode);
-  }
```

Comment:
@travissanderson-wf this code was causing the empty branches by adding a new `LeafNode` to the parent _during removal_ which escaped the parent's `childrenToRemove` logic. This logic seems unnecessary except in the case of converting the root to a leaf which is handled separated by `RTree._resetRoot`, so I can simply remove it.